kafka_consumer_freshness_tracker/src/main/java/com/tesla/data/consumer/freshness/ConsumerFreshness.java

```diff
@@ -113,22 +123,73 @@ public static void main(String[] args) throws IOException, InterruptedException
   }
 
   private void setup(Map<String, Object> conf) {
-    this.burrow = new Burrow((Map<String, Object>) conf.get("burrow"));
+    setupWithBurrow(conf, new Burrow((Map<String, Object>) conf.get("burrow")));
+  }
+
+  @VisibleForTesting
+  void setupWithBurrow(Map<String, Object> conf, Burrow burrow) {
+    this.burrow = burrow;
     int workerThreadCount = (int) conf.getOrDefault("workerThreadCount", DEFAULT_WORKER_THREADS);
     this.availableWorkers = ((List<Map<String, Object>>) conf.get("clusters")).stream()
         .map(clusterConf -> {
+          // validate the cluster configuration
+          Optional<String> validationErrorMsg = validateClusterConf(clusterConf);
+          if (validationErrorMsg.isPresent()) {
+            String msg = String.format("configuration for cluster %s is invalid: %s",
+                    clusterConf.get("name"),
+                    validationErrorMsg.get()
+            );
+            if (strict) {
+              throw new RuntimeException(msg);
+            } else {
+              LOG.warn(msg);
+            }
+          }
           // allow each cluster to override the number of workers, if desired
           int numConsumers = (int) clusterConf.getOrDefault("numConsumers", DEFAULT_KAFKA_CONSUMER_COUNT);
           ArrayBlockingQueue<KafkaConsumer> queue = new ArrayBlockingQueue<>(numConsumers);
           for (int i = 0; i < numConsumers; i++) {
             queue.add(createConsumer(clusterConf));
           }
           return new AbstractMap.SimpleEntry<>((String) clusterConf.get("name"), queue);
-        }).collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+        })
+        .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
 
     this.executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(workerThreadCount));
   }
 
+  private Optional<String> validateClusterConf(Map<String, Object> clusterConf) {
+    final String clusterName = (String) clusterConf.get("name");
+    final Set<String> bootstrapServersFromBurrow;
+    try {
+      bootstrapServersFromBurrow = new HashSet<>(this.burrow.getClusterBootstrapServers(clusterName));
+    } catch (IOException e) {
+      this.metrics.burrowClusterDetailReadFailed.inc();
+      return Optional.of(String.format("failed to read cluster detail from Burrow: %s", e));
```

Comment:
i think you want to just do `+ e.getMessage()` rather than formatting. Formatting doesn't save you runtime case here

Reply:
> doesn't save you runtime case 
what do you mean by that?

Reply:
went ahead and make the change, but would still be interested to know the implications of using format vs string concat here

Reply:
would `return Optional.of(String.format("failed to read cluster detail from Burrow: %s", e.getMessage()));` work?